plutus-core/plutus-core/src/PlutusCore/Pretty/ConfigName.hs

```diff
@@ -1,125 +1,36 @@
-        , _prettyConfigPlcLoop :: PrettyConfigPlcLoop
-        }
-
-i.e. there is a 'PrettyConfigName' at the current level, but you can descend further and there
-will be a a 'PrettyConfigName' as well. While this might work, we're not in the Inception movie
-and hence we define
-
-    instance PrettyBy (PrettyConfigClassic configName) (tyname a) =>
-            PrettyBy (PrettyConfigClassic configName) (Type tyname a)
-
-i.e. require that a @tyname a@ must be pretty-printable with the same config as an entire 'Type'.
-
-... and immediately run into the O(n * m) number of instances problem:
-
-    [Classic, Refined] x [Name, TyName, NameWithType, TyNameWithKind]
-
-where @[Classic, Refined]@ are pretty-printing strategies (we can add more in future) and
-@[Name, TyName, NameWithType, TyNameWithKind]@ are PLC names (we will likely add more in future).
-We do not need this level of extensibility (pretty-printing names differently depending on a
-pretty-printing strategy used), so we do the following twist: for any pretty-printing strategy
-we require that it must contain a PLC names pretty-printing config and then define a single instance
-for each of the PLC names. E.g. for 'Name' it looks like this:
-
-    instance HasPrettyConfigName config => PrettyBy config (Name ann) where
-
-i.e. "you can pretty-print a 'Name' using any config as long as a 'PrettyConfigName' can be
-extracted from it". This results in O(n + m) number of instances, with 'HasPrettyConfigName'
-instances being defined like
-
-    instance configName ~ PrettyConfigName => HasPrettyConfigName (PrettyConfigClassic configName) where
-        toPrettyConfigName = _pccConfigName
-
-Here we also hardcode the PLC names pretty-printing config to be 'PrettyConfigName' which sometimes
-contains redundant information (e.g. to pretty-print a 'Name' the '_pcnShowsAttached' field is not
-required). This is something that we may try to improve later.
--}
-
 -- | A config that determines how to pretty-print a PLC name.
 newtype PrettyConfigName = PrettyConfigName
-    { _pcnShowsUnique :: Bool -- ^ Whether to show the 'Unique' of a name or not.
-    }
+  { _pcnShowsUnique :: Bool
+  -- ^ Whether to show the 'Unique' of a name or not.
+  }
 
 type instance HasPrettyDefaults PrettyConfigName = 'True
 
 -- | A class of configs from which a 'PrettyConfigName' can be extracted.
 class HasPrettyConfigName config where
-    toPrettyConfigName :: config -> PrettyConfigName
+  toPrettyConfigName :: config -> PrettyConfigName
 
 instance HasPrettyConfigName (Sole PrettyConfigName) where
-    toPrettyConfigName = coerce
+  toPrettyConfigName = coerce
 
 -- | The 'PrettyConfigName' used by default: don't print 'Unique's.
 defPrettyConfigName :: PrettyConfigName
-defPrettyConfigName = PrettyConfigName False
+defPrettyConfigName = PrettyConfigName True
 
 -- | The 'PrettyConfigName' used for debugging: print 'Unique's.
 debugPrettyConfigName :: PrettyConfigName
```

Comment:
This should get flipped as well then (otherwise we end up having two identical configs and not a single other one for pretty-printing names), so that one can pretty-print stuff without uniques. The name should be changed too as it's no longer relevant to any debugging.

Reply:
👌🏼